src/tplr/comms.py

```diff
@@ -94,7 +94,13 @@ def __init__(
         self.save_location = os.path.join("/tmp", f"hotkey_{hotkey}")
         os.makedirs(self.save_location, exist_ok=True)
         self.key_prefix = key_prefix
+
+        ## a single aiobotocore session and a dictionary of clients
         self.session = get_session()
+        self._s3_clients: Dict[
+            tuple, Any
```

Comment:
is the `tuple` correctly spelled? shouldn't it be `Tuple` from `Typings`?

Reply:
instead of Any, add the S3_client class. that you get from the session.create_client

Reply:
Updated in [0bfb4f1](https://github.com/tplr-ai/templar/pull/181/commits/0bfb4f1e3a8466bb06615e4d16c8b46dd9959efc)

Reply:
it is just the standard collection types. `list` instead of `typings.List`, `dict` instead of `typing.Dict` and so on. Also, in python 3.10 or something we can do `str | None` instead of `typing.Optional[str]` or `int | str` instead of `typing.Untion[int, str]`

Reply:
old habits die hard :)